src/Agent.Sdk/Util/IOUtil.cs

```diff
@@ -337,18 +330,21 @@ public static string ResolvePath(String rootPath, String relativePath)
                     throw new InvalidOperationException($"The file path {relativePath} is invalid");
                 }
 
-#if OS_WINDOWS
-                if (segments.Count > 1)
+                if (PlatformUtil.RunningOnOS == PlatformUtil.OS.Windows)
                 {
-                    return String.Join(Path.DirectorySeparatorChar, segments);
+                    if (segments.Count > 1)
+                    {
+                        return String.Join(Path.DirectorySeparatorChar, segments);
+                    }
+                    else
+                    {
+                        return segments.Pop() + Path.DirectorySeparatorChar;
+                    }
```

Comment:
this doesn't look right. If we <= 0 items we pop? did the UI mess up?

Reply:
I didn't touch this code; it got reindented since it's wrapped in an if block now. Something for us to investigate down the line.